Sources/Benchmark/BenchmarkCommand.swift

```diff
@@ -29,17 +29,32 @@ internal struct BenchmarkCommand: ParsableCommand {
     @Option(help: "Number of warm-up iterations to run.")
     var warmupIterations: Int?
 
+    @Option(help: "Minimal time to run when automatically detecting number iterations.")
+    var minTime: Double?
+
+    @Option(
+        help: "Maximum number of iterations to run when automatically detecting number iterations.")
+    var maxIterations: Int?
+
     var settings: [BenchmarkSetting] {
         var result: [BenchmarkSetting] = []
-        if filter != nil {
-            result.append(.filter(filter!))
+
+        if let value = filter {
+            result.append(.filter(value))
+        }
+        if let value = iterations {
+            result.append(.iterations(value))
+        }
+        if let value = warmupIterations {
+            result.append(.warmupIterations(value))
         }
-        if iterations != nil {
-            result.append(.iterations(iterations!))
+        if let value = minTime {
+            result.append(.minTime(value))
         }
-        if warmupIterations != nil {
-            result.append(.warmupIterations(warmupIterations!))
+        if let value = maxIterations {
+            result.append(.maxIterations(value))
         }
+
```

Comment:
This code seems somewhat repetitive. I wonder if we could come up with a way to write this once, and have it be generic across a variety of fields and data types... one might call it _datatype generic programming_. WDYT?

Reply:
That *datatype generic programming* thing sure sounds great.